configs/config.toml

```diff
@@ -80,18 +65,47 @@ max_file_count = 3
 # - compression(optional): an enum values to determine what compression the serializer
 #   should perform on the data transiting through the stream. Currently supported
 #   compression schemes are Lz4 and Disabled. Defaults to Disabled.
+# - persistence(optional): helps persist relevant information for data recovery purposes,
+#   used when there is a network/system failure. 
 #
 # In the following config for the device_shadow stream we set buf_size to 1. streams is
 # internally constructed as a map of Name -> Config
 [streams.device_shadow]
 topic = "/tenants/{tenant_id}/devices/{device_id}/device_shadow/jsonarray"
 buf_size = 1
 
+# Example using compression
 [streams.imu]
 topic = "/tenants/{tenant_id}/devices/{device_id}/events/imu/jsonarray"
 buf_size = 100
 compression = "Lz4"
 
+# Configuration details associated with uplink's persistent storage module which writes publish
+# packets to disk in case of slow or crashed network, for recovery purposes.
+# 
+# Required Parameters
+# - max_file_size(defaults to 100MB): Maximum size upto which single persistence file can grow.
+#   This value alone can be changed to configure in-memory storage of packets during network recovery.
+# - path(optional): Path to directory for storing backlog in files, shouldn't contain anything else.
+#   Please ensure the location is unique for each stream to ensure there is no clash in files. 
+# - max_file_count(optional, defaults to 3): Maximum number of persistence files allowed on disk.
+#
+# NOTE: Persitence is an optional feature that is disabled by default, i.e. if not inlcuded in the 
+# configuration, we use transient(in-memory) storage to handle network downtime only.
+[streams.gps]
+topic = "/tenants/{tenant_id}/devices/{device_id}/events/gps/jsonarray"
+buf_size = 10
+persistence = { path = "/tmp/uplink/gps", max_file_size = 1048576, max_file_count = 10 }
```

Comment:
This will be problematic if same path is used in other stream's config. We should only specify base path here. Serializer init should create folder with `stream` name in base path

Reply:
This is exactly what I had called to ask you about, would the customer expect the behavior of creating a sub-directory or not.